### DISCUSSION TITLE
generate hologram and reconstruct

### DISCUSSION THREAD

**Discussion by emanuelevivoli** [rafael-fuente/Diffraction-Simulations--Angular-Spectrum-Method#38] [Q&A]
Hello, and compliments for the library, it's Amazing!

My question is if it's possible to **generate the hologram and reconstruct the aperture**.
The only examples that seems to do what I'm aiming are `phase_hologram_github_logo` and `phase_hologram_snowflake`, in which you generate the hologram, and reconstruct the aperture. 
The only thing is that you are using FourierPhaseRetrieval to create your phase_mask. If I use the same approach with my hologram (already collected) as `phase_mask`, I'm not able to reconstruct the object that at distance z created that hologram.

Hope you can help me,
Thanks

**Comment by mdhoney** [comment]
I'm interested in this too.  Do you have your `phase_mask` in the correct format?  [In another post](https://github.com/rafael-fuente/diffractsim/discussions/8#discussioncomment-1896397) the author states it needs to be encoded as HSV, not grayscale.
